consent-ui/src/app/ais/sca-select-page/sca-select-page.component.ts

```diff
@@ -1,48 +1,74 @@
-import { Component, OnInit } from '@angular/core';
-import {ConsentUtil} from "../common/consent-util";
-import {ApiHeaders} from "../../api/api.headers";
-import {AisConsentToGrant} from "../common/dto/ais-consent";
+import {Component, OnInit} from '@angular/core';
 import {ActivatedRoute} from "@angular/router";
+import {FormBuilder, FormGroup, Validators} from "@angular/forms";
+import {ConsentAuthorizationService, ScaUserData} from "../../api";
 import {SessionService} from "../../common/session.service";
-import {ConsentAuthorizationService} from "../../api";
+import {ApiHeaders} from "../../api/api.headers";
+import {StubUtil} from "../common/stub-util";
+import {Subscription} from "rxjs";
 
 @Component({
-  selector: 'consent-app-sca-select-page',
-  templateUrl: './sca-select-page.component.html',
-  styleUrls: ['./sca-select-page.component.scss']
+    selector: 'consent-app-sca-select-page',
+    templateUrl: './sca-select-page.component.html',
+    styleUrls: ['./sca-select-page.component.scss']
 })
 export class ScaSelectPageComponent implements OnInit {
-  private redirectTo: string;
-  private authorizationId: string;
-  private aisConsent: AisConsentToGrant;
 
-  constructor(
-    private activatedRoute: ActivatedRoute,
-    private sessionService: SessionService,
-    private consentAuthorisation: ConsentAuthorizationService
-  ) {
-  }
+    authorizationSessionId: string = '';
+    redirectCode: string = '';
+    scaMethodForm: FormGroup;
+    scaMethods: ScaUserData[];
+    private subscriptions: Subscription[] = [];
+
+    constructor(private sessionService: SessionService,
+                private consentAuthorizationService: ConsentAuthorizationService,
+                private route: ActivatedRoute,
+                private formBuilder: FormBuilder) {
+    }
+
+    ngOnInit() {
+        this.initialScaMethodForm();
+        this.authorizationSessionId = this.route.parent.snapshot.paramMap.get("authId");
+        this.redirectCode = this.sessionService.getRedirectCode(this.authorizationSessionId);
+        this.loadAvailableMethods();
+    }
 
-  ngOnInit() {
-    this.activatedRoute.parent.params.subscribe(res => {
-      this.authorizationId = res.authId;
-      this.aisConsent = ConsentUtil.getOrDefault(this.authorizationId, this.sessionService);
-      this.loadRedirectUri();
-    });
-  }
+    onSubmit(): void {
+        this.subscriptions.push(
+            this.consentAuthorizationService.embeddedUsingPOST(
+                this.authorizationSessionId,
+                StubUtil.X_REQUEST_ID, // TODO: real values instead of stubs
+                StubUtil.X_XSRF_TOKEN, // TODO: real values instead of stubs
+                this.redirectCode,
+                {scaAuthenticationData: {SCA_CHALLENGE_ID: this.scaMethodForm.get('selectedMethodValue').value}},
```

Comment:
Instead of having string literal `selectedMethodValue`, you can consider having `selectedMethod: FormControl`